app/src/main/java/org/simple/clinic/sms/SmsReaderImpl.kt

```diff
@@ -0,0 +1,10 @@
+package org.simple.clinic.sms
+
+import io.reactivex.Single
+import org.simple.clinic.activity.TheActivity
+import javax.inject.Inject
+
+class SmsReaderImpl @Inject constructor(private val activity: TheActivity) : SmsReader {
+
+  override fun waitForSms() = Single.just(SmsReadResult.SmsReadSuccess("") as SmsReadResult)!!
```

Comment:
I *think* this function name can be improved because we never use the term `wait` for asynchronous calls. But I'm sure you will also realize this once you use this function.

Reply:
Right. I'll revisit the naming once the SMS read functionality is complete and see if there can be a more appropriate name.